src/test/java/uk/gov/companieshouse/api/testdata/service/impl/AcspProfileServiceImplTest.java

```diff
@@ -83,6 +92,142 @@ void createAcspProfileWithDefaultValues() throws DataException {
+        AcspProfileSpec spec = new AcspProfileSpec();
+        spec.setStatus("active");
+        spec.setType("ltd");
+
+        // Creating multiple AML details entries
+        AmlSpec amlSpec1 = new AmlSpec();
+        amlSpec1.setSupervisoryBody("Supervisory Body 1");
+        amlSpec1.setMembershipDetails("Membership Details 1");
+
+        AmlSpec amlSpec2 = new AmlSpec();
+        amlSpec2.setSupervisoryBody("Supervisory Body 2");
+        amlSpec2.setMembershipDetails("Membership Details 2");
+
+        AmlSpec amlSpec3 = new AmlSpec();
+        amlSpec3.setSupervisoryBody("Supervisory Body 3");
+        amlSpec3.setMembershipDetails("Membership Details 3");
+
+        spec.setAmlDetails(List.of(amlSpec1, amlSpec2, amlSpec3));
+        return spec;
+    }
+
+    @Test
+    void createAcspProfileWithNullAmlDetails() throws DataException {
+        AcspProfileSpec spec = new AcspProfileSpec();
+        spec.setStatus("active");
+        spec.setType("ltd");
+        spec.setAmlDetails(null); // Setting AmlDetails as null
+
+        when(randomService.getString(8)).thenReturn("randomId");
+        AcspProfile savedProfile = createSavedProfile();
+
+        when(repository.save(any(AcspProfile.class))).thenReturn(savedProfile);
+
+        AcspProfileData result = service.create(spec);
+
+        assertNotNull(result);
+        assertEquals("randomId", result.getAcspNumber());
+
+        ArgumentCaptor<AcspProfile> captor = ArgumentCaptor.forClass(AcspProfile.class);
+        verify(repository).save(captor.capture());
+
+        AcspProfile captured = captor.getValue();
+        assertNotNull(captured);
+        assertEquals("randomId", captured.getId());
+        assertEquals("randomId", captured.getAcspNumber());
+        assertEquals(spec.getStatus(), captured.getStatus()); // Default value
+        assertEquals(spec.getType(), captured.getType()); // Default value
+        assertEquals("Test Data Generator randomId Company Ltd", captured.getName());
+        assertEquals("/authorised-corporate-service-providers/randomId", captured.getLinksSelf());
+        assertNull(captured.getAmlDetails()); // Ensure it remains null
+    }
+
+    @Test
+    void createAcspProfileWithEmptyAmlDetails() throws DataException {
+        AcspProfileSpec spec = new AcspProfileSpec();
+        spec.setStatus("active");
+        spec.setType("ltd");
+        spec.setAmlDetails(Collections.emptyList()); // Setting an empty list
+
+        when(randomService.getString(8)).thenReturn("randomId");
+        AcspProfile savedProfile = createSavedProfile();
+        savedProfile.setAmlDetails(new ArrayList<>()); // Ensure it's an empty list
+
+        when(repository.save(any(AcspProfile.class))).thenReturn(savedProfile);
+
+        AcspProfileData result = service.create(spec);
+
+        assertNotNull(result);
+        assertEquals("randomId", result.getAcspNumber());
+
+        ArgumentCaptor<AcspProfile> captor = ArgumentCaptor.forClass(AcspProfile.class);
```

Comment:
We are repeating this in many methods, may I suggest creating a new instance field instead

```
    @Captor
    private ArgumentCaptor<AcspProfile> profileCaptor;
```